packages/search-in-workspace/src/browser/search-in-workspace-frontend-module.ts

```diff
@@ -5,14 +5,34 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-import { ContainerModule } from "inversify";
+import { ContainerModule, interfaces } from "inversify";
 import { SearchInWorkspaceService, SearchInWorkspaceClientImpl } from './search-in-workspace-service';
 import { SearchInWorkspaceServer } from '../common/search-in-workspace-interface';
-import { WebSocketConnectionProvider, KeybindingContribution } from '@theia/core/lib/browser';
+import { WebSocketConnectionProvider, KeybindingContribution, WidgetFactory, createTreeContainer, TreeWidget } from '@theia/core/lib/browser';
 import { QuickSearchInWorkspace, SearchInWorkspaceContributions } from './quick-search-in-workspace';
-import { CommandContribution, MenuContribution } from "@theia/core";
+import { CommandContribution, MenuContribution, ResourceResolver } from "@theia/core";
+import { SearchInWorkspaceWidget } from "./search-in-workspace-widget";
+import { SearchInWorkspaceResultTreeWidget } from "./search-in-workspace-result-tree-widget";
+import { SearchInWorkspaceFrontendContribution } from "./search-in-workspace-frontend-contribution";
+import { MemoryTextResourceResolver } from "./memory-text-resource";
+
+import "../../src/browser/styles/index.css";
 
 export default new ContainerModule(bind => {
+    bind(SearchInWorkspaceWidget).toSelf();
+    bind<WidgetFactory>(WidgetFactory).toDynamicValue(ctx => ({
+        id: SearchInWorkspaceWidget.ID,
+        createWidget: () => ctx.container.get(SearchInWorkspaceWidget)
+    }));
+    bind(SearchInWorkspaceResultTreeWidget).toDynamicValue(ctx => createSearchTreeWidget(ctx.container));
+
+    bind(SearchInWorkspaceFrontendContribution).toSelf().inSingletonScope();
+    for (const identifier of [CommandContribution, MenuContribution, KeybindingContribution]) {
+        bind(identifier).toDynamicValue(ctx =>
```

Comment:
fyi: `toService`

Reply:
Ok, we should do this on other code places as well, then.